libs3/src/object.c

```diff
@@ -569,28 +570,37 @@ static S3Status getObjectAttributesXmlCallback(const char* elementPath, const ch
 		return S3StatusOK;
 	}
 
-	if (!strcmp(elementPath, "GetObjectAttributesResponse/Checksum/ChecksumCRC32")) {
+	static const char crc32_xml_path[] = "GetObjectAttributesResponse/Checksum/ChecksumCRC32";
+	static const char crc32c_xml_path[] = "GetObjectAttributesResponse/Checksum/ChecksumCRC32C";
+	static const char crc64nvme_xml_path[] = "GetObjectAttributesResponse/Checksum/ChecksumCRC64NVME";
+	static const char sha1_xml_path[] = "GetObjectAttributesResponse/Checksum/ChecksumSHA1";
+	static const char sha256_xml_path[] = "GetObjectAttributesResponse/Checksum/ChecksumSHA256";
+	static const char checksum_type_xml_path[] = "GetObjectAttributesResponse/Checksum/ChecksumType";
+	static const char storage_class_xml_path[] = "GetObjectAttributesResponse/StorageClass";
+	static const char object_size_xml_path[] = "GetObjectAttributesResponse/ObjectSize";
+
+	if (0 == strncmp(elementPath, crc32_xml_path, strlen(crc32_xml_path))) {
 		string_buffer_append(goaData->checksumCRC32, data, dataLen, fit);
 	}
-	if (!strcmp(elementPath, "GetObjectAttributesResponse/Checksum/ChecksumCRC32C")) {
+	else if (0 == strncmp(elementPath, crc32c_xml_path, strlen(crc32c_xml_path))) {
```

Comment:
Just to confirm... `elementPath` cannot be equal to more than one of these at a time, right?